scripts/stackstorm.js

```diff
@@ -370,30 +373,26 @@ module.exports = function(robot) {
       }
       postDataHandler.postData(data);
 
-      res.send('{"status": "completed", "msg": "Message posted successfully"}');
-    } catch (e) {
-      robot.logger.error("Unable to decode JSON: " + e);
-      robot.logger.error(e.stack);
-      res.send('{"status": "failed", "msg": "An error occurred trying to post the message: ' + e + '"}');
+      res.send(JSON.stringify({
+        "status": "completed",
+        "msg": "Message posted successfully"
+      }));
+    } catch (err) {
+      logErrorAndExit(err, res)
     }
   });
 
   var commands_load_interval;
 
   function start() {
-    robot.error(hubotErrorCallback);
+    robot.error(logErrorAndExit);
 
     api_client.stream.listen().catch(function (err) {
       exitProcessWithLog('Unable to connect to stream: ', err);
     }).then(function (source) {
       source.onerror = function (err) {
-        // TODO: squeeze a little bit more info out of evensource.js
-        if (err.status === 401) {
-          robot.logger.error('Stream error:', err);
-        } else {
-          var error_message = util.format('stream error: [type: %s; status: %s]', err.type, err.status);
-          exitProcessWithLog(error_message, null);
-        }
```

Comment:
@blag Sorry for this later comment. Just tested new `stackstorm.js` code and realized by removing this part of code, the user has no idea what is issue when an invalid `ST2_API_KEY` or `ST2_AUTH_TOKEN` is encountered.  Only error message is  `ERROR undefined`. Before this change, it will get more meaningful messages  : 
```
For invalid ST2_AUTH_TOKEN:
[Wed Jun 05 2019 17:31:31 GMT+0000 (Coordinated Universal Time)] ERROR Stream error: Event { type: 'error', status: 401 }
[Wed Jun 05 2019 17:31:31 GMT+0000 (Coordinated Universal Time)] ERROR Failed to retrieve commands from https://localhost/api Unauthorized
[Wed Jun 05 2019 17:31:31 GMT+0000 (Coordinated Universal Time)] ERROR undefined

For invalid ST2_APY_KEY:
[Wed Jun 05 2019 17:38:28 GMT+0000 (Coordinated Universal Time)] ERROR Stream error: Event { type: 'error', status: 401 }
[Wed Jun 05 2019 17:38:28 GMT+0000 (Coordinated Universal Time)] ERROR Failed to retrieve commands from https://localhost/api Unauthorized - ApiKey with 
key_hash=cddbf16c36e281d74efd98b8fc9467d0b68e304ae4b0e0026f9b8ec5a95043fef7393dddd2edb712c4ed499f4752f7e233991b8b9053958748a1be051c30a1a5 not found.
[Wed Jun 05 2019 17:38:28 GMT+0000 (Coordinated Universal Time)] ERROR undefined
```

Reply:
Are you sure that was where that error message was coming from? Because the `error_message` variable should have been printing something like `stream error: [type: %s; status: %s]`, but the error message was different: `Stream error: Event { type: 'error', status: 401 }`.

Reply:
Line 394  `util.format('stream error: [type: %s; status: %s]', err.type, err.status);`  printed out `Stream error: Event { type: 'error', status: 401 }`.

Reply:
I've updated this to not exit immediately, and to simply emit an informative error about what's going on. The EventSource object should attempt to reconnect automatically.